src/main/java/org/sagebionetworks/web/client/widget/entity/menu/v3/EntityActionMenuImpl.java

```diff
@@ -135,14 +137,25 @@ public void hideAllActions() {
     synchronizeView();
   }
 
+  @Override
+  public void addPropUpdateListener(Consumer<EntityActionMenuProps> listener) {
```

Comment:
Every time the action menu props change, we need to re-render the EntityTitleBar and pass along these props, so we'll add a way to register listeners here.

Reply:
Any possibility of stale listeners?  I guess the title bar widget is shared across entities, so this is probably not an issue.

Reply:
This implementation causes listeners that reference the same BasicTitleBar widget to accumulate. Since there only one instance of the BasicTitleBar widget on the page at a time, and that is the only widget that listens to the action menu, I converted this to only store one listener in state at a time. 

It's not perfect; there might still be memory leak issues, but it's less likely. I'm placing a bet that we will have another chance to re-architect or simplify this before we would need to track multiple listeners.